### PR TITLE
MGMT-12332: Kube API subsystem tests for kernel arguments

### PR DESCRIPTION
**Pull request by ori-amizur**
## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [x] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md
/cc @filanov

### CODE REVIEW

File: subsystem/kubeapi_test.go [hunk 1]
```diff
@@ -748,6 +749,22 @@ func cleanUpCRs(ctx context.Context, client k8sclient.Client) {
 	}
 }
 
+func kubeKernelArgs(internalInfraEnv *common.InfraEnv) (ret []v1beta1.KernelArgument) {
```
Comment (filanov) [inline subsystem/kubeapi_test.go:752]:
```suggestion
func extractKernelArgs(internalInfraEnv *common.InfraEnv) (ret []v1beta1.KernelArgument) {
```

File: subsystem/kubeapi_test.go [hunk 2]
```diff
@@ -2962,6 +2979,120 @@ var _ = Describe("[kube-api]cluster installation", func() {
...
+				infraEnv := getInfraEnvCRD(ctx, kubeClient, infraEnvKey)
+				infraEnv.Spec.KernelArguments = kargs
+				return kubeClient.Update(ctx, infraEnv)
+			}, "10s", "1s").ShouldNot(HaveOccurred())
+			Eventually(func(g Gomega) {
+				infraEnv := getInfraEnvCRD(ctx, kubeClient, infraEnvKey)
+				g.Expect(expectedSpecKargs(infraEnv.Spec.KernelArguments)).To(Equal(kargs))
+				dbInfraEnv := getInfraEnvFromDBByKubeKey(ctx, db, infraEnvKey, 1)
+				g.Expect(kubeKernelArgs(dbInfraEnv)).To(Equal(kargs))
+			}, "30s", "2s").Should(Succeed())
```
Comment (filanov) [inline subsystem/kubeapi_test.go:3028]: i sec is enough.
```suggestion
			}, "30s", "1s").Should(Succeed())
```

File: subsystem/kubeapi_test.go [hunk 3]
```diff
@@ -2962,6 +2979,120 @@ var _ = Describe("[kube-api]cluster installation", func() {
...
+				}
+				deployInfraEnvCRD(ctx, kubeClient, infraNsName.Name, infraEnvSpec)
+				infraEnv := getInfraEnvCRD(ctx, kubeClient, infraEnvKey)
+				infraEnv.Spec.KernelArguments = []v1beta1.KernelArgument{
+					{
+						Operation: operation,
+						Value:     value,
+					},
+				}
+				Expect(kubeClient.Update(ctx, infraEnv)).To(HaveOccurred())
```
Comment (filanov) [inline subsystem/kubeapi_test.go:3089]: what are you checking here? 
do you have webhooks that are validating the parameters? 
unless there are webhooks the api should not return an error, reconcile will detect the error and reflect it in the sync condition

File: subsystem/kubeapi_test.go (hunk 1 above)
Comment (ori-amizur) [reply subsystem/kubeapi_test.go:752]: OK

File: subsystem/kubeapi_test.go (hunk 3 above)
Comment (ori-amizur) [reply subsystem/kubeapi_test.go:3089]: The API defines enums and pattern, so it should be rejected by the API.
Please see https://github.com/openshift/assisted-service/blob/4efed6128e9e854c41a69b262b9374c5768fcf3b/api/v1beta1/infraenv_types.go#L108-L119

Comment (ori-amizur) [reply subsystem/kubeapi_test.go:3089]: Anyway, the API returns error in these cases.

**Review by filanov — APPROVED** [review]

**Pull request merged**